x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -1615,6 +1618,23 @@ private LogicalPlan rule(Eval eval) {
         }
     }
 
+    /**
+     * Replace type converting eval with aliasing eval when type change does not occur.
+     * A following ReplaceAliasingEvalWithProject will effectively convert ReferenceAttribute into FieldAttribute,
```

Comment:
super-duper nit: Could add an example, and could use `@link` in the javadoc to prevent this from getting stale if we rename/refactor our optimizer rules.
```suggestion
     * A following {@link ReplaceAliasingEvalWithProject} will effectively convert {@link ReferenceAttribute} into {@link FieldAttribute},
```